### PR TITLE
task done1.

### PR DESCRIPTION
**Pull request by AndrewOliferchuk**

### CODE REVIEW

File: src/main/java/com/example/demo/controller/BookController.java [hunk 1]
```diff
@@ -33,12 +34,13 @@ public BookDto getBookById(@PathVariable Long id) {
...
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
+    public BookDto updateBook(
+            @PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/BookController.java]:
```suggestion
            @PathVariable @Positive Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {
```
You could also add validation for id. If so don't forget to add `@Validated` under the controller

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java]: For String is better to use `@NotBlank`. Check other places

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
     private String title;
+    @NotNull
     private String author;
+    @ISBN
     private String isbn;
+    @NotNull
+    @Min(value = 0)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java]:
```suggestion
    @Positive
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/com/example/demo/controller/BookController.java [hunk 4]
```diff
@@ -32,13 +35,16 @@ public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
+    @Validated
```
Comment (koval-kn) [inline src/main/java/com/example/demo/controller/BookController.java]: why do you use it? `@Valid` is enough

**Review by koval-kn — APPROVED** [review]

**Pull request merged**